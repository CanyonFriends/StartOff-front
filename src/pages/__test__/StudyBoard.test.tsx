/**
 * @jest-environment jsdom
 */
import React from 'react';
import { waitFor } from '@testing-library/react';
import { render } from '../../test-utils';
import StudyBoardPage from '../StudyBoard';

beforeEach(() => {
  jest.resetAllMocks();
});

describe('<StudyBoard> 페이지', () => {
  it('snapshot 체크', async () => {
    const component = render(<StudyBoardPage />);
    await waitFor(() => expect(component.container).toMatchSnapshot());
  });

  it('렌더링 테스트', async () => {
    const component = render(<StudyBoardPage />);

    component.getByText('글 쓰기');
    component.getByText('검색');
  });
});
