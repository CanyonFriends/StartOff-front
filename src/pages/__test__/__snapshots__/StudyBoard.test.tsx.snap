// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StudyBoard> 페이지 snapshot 체크 1`] = `
<div>
  <div
    id="modal"
  />
  <header
    class="css-15jvef"
  >
    <div
      aria-label=""
      class="css-y8pvf0"
    >
      <div
        aria-label=""
        class="css-ximz1h"
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 0c-6.626 0-12 5.373-12 12 0 5.302 3.438 9.8 8.207 11.387.599.111.793-.261.793-.577v-2.234c-3.338.726-4.033-1.416-4.033-1.416-.546-1.387-1.333-1.756-1.333-1.756-1.089-.745.083-.729.083-.729 1.205.084 1.839 1.237 1.839 1.237 1.07 1.834 2.807 1.304 3.492.997.107-.775.418-1.305.762-1.604-2.665-.305-5.467-1.334-5.467-5.931 0-1.311.469-2.381 1.236-3.221-.124-.303-.535-1.524.117-3.176 0 0 1.008-.322 3.301 1.23.957-.266 1.983-.399 3.003-.404 1.02.005 2.047.138 3.006.404 2.291-1.552 3.297-1.23 3.297-1.23.653 1.653.242 2.874.118 3.176.77.84 1.235 1.911 1.235 3.221 0 4.609-2.807 5.624-5.479 5.921.43.372.823 1.102.823 2.222v3.293c0 .319.192.694.801.576 4.765-1.589 8.199-6.086 8.199-11.386 0-6.627-5.373-12-12-12z"
          />
        </svg>
      </div>
      <ul
        class="css-foxkse"
      >
        <li
          class="css-6xh7y2"
        >
          <div
            aria-label=""
            class="css-147wqve"
          >
            <a
              href="/board/study"
            >
              스터디
            </a>
          </div>
        </li>
        <li
          class="css-6xh7y2"
        >
          <div
            aria-label=""
            class="css-147wqve"
          >
            <a
              href="/"
            >
              게시판2
            </a>
          </div>
        </li>
        <li
          class="css-6xh7y2"
        >
          <div
            aria-label=""
            class="css-147wqve"
          >
            <a
              href="/"
            >
              게시판3
            </a>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="css-12x73ce"
    >
      <div
        aria-label=""
        class="css-wysmox"
      >
        <a
          href="/signin"
        >
          <div
            aria-label=""
            class="css-1q39o9r"
          >
            <svg
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 9v-4l8 7-8 7v-4h-8v-6h8zm6-7c-1.787 0-3.46.474-4.911 1.295l.228.2 1.395 1.221c1.004-.456 2.115-.716 3.288-.716 4.411 0 8 3.589 8 8s-3.589 8-8 8c-1.173 0-2.284-.26-3.288-.715l-1.395 1.221-.228.2c1.451.82 3.124 1.294 4.911 1.294 5.522 0 10-4.477 10-10s-4.478-10-10-10z"
              />
            </svg>
          </div>
        </a>
      </div>
    </div>
  </header>
  <div
    class="css-45cbny"
  >
    <header
      class="css-z0wawm"
    >
      <div
        class="css-t8n9tm"
      >
        <div
          aria-label=""
          class="css-gd9wkw"
        >
          <button
            class="css-1fbvvo9"
            type="button"
            width="fit-content"
          >
            제목
          </button>
        </div>
        <form
          class="css-1es7re"
        >
          <input
            aria-label="search-input"
            class="css-vcix8v"
            id="search-input"
            name=""
            placeholder="검색어를 입력해주세요"
            type="text"
            value=""
            width="80%"
          />
          <button
            class="css-1fbvvo9"
            type="submit"
            width="fit-content"
          >
            검색 
            <div
              aria-label="search-icon"
              class="css-v4cn5l"
            >
              <svg
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M23.111 20.058l-4.977-4.977c.965-1.52 1.523-3.322 1.523-5.251 0-5.42-4.409-9.83-9.829-9.83-5.42 0-9.828 4.41-9.828 9.83s4.408 9.83 9.829 9.83c1.834 0 3.552-.505 5.022-1.383l5.021 5.021c2.144 2.141 5.384-1.096 3.239-3.24zm-20.064-10.228c0-3.739 3.043-6.782 6.782-6.782s6.782 3.042 6.782 6.782-3.043 6.782-6.782 6.782-6.782-3.043-6.782-6.782zm2.01-1.764c1.984-4.599 8.664-4.066 9.922.749-2.534-2.974-6.993-3.294-9.922-.749z"
                />
              </svg>
            </div>
          </button>
        </form>
      </div>
      <button
        class="css-1lwhfay"
        type="submit"
        width="fit-content"
      >
        글 쓰기
      </button>
    </header>
    <ul
      class="css-1h10yx"
    />
  </div>
</div>
`;
